### PR TITLE
feat: Themes

### PR DESCRIPTION
**Pull request by ariansobczak-rst**
Introduced Theme generator, dark and light theme for AdminJS.

**Dark theme**
image
**Light theme**
image

To use themes in AdminJS just import from `@adminjs/themes` or create using ThemeGenerator
```ts
const theme = new ThemeGenerator({
{
  details: { name: 'AdminJS theme' },
  colors: {
    bg: '#FFFFFF'
  }
})
// Then export to use as a theme in app as an AdminJSTheme
theme.toJSON()
```
or import from themes
```ts
import { DarkTheme } from '@adminjs/themes';

export const adminConfig: AdminJSOptions = {
  branding: {
    theme: DarkTheme
  }
};
```

Related with https://github.com/SoftwareBrothers/adminjs/pull/1025

Palete page
image
image

### CODE REVIEW

File: src/theme-generator.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import { colors as defaultColors } from '@adminjs/design-system';
+import { AdminJSTheme } from 'adminjs';
```
Comment (ariansobczak-rst) [inline src/theme-generator.ts:2]: Added in https://github.com/SoftwareBrothers/adminjs/pull/1025

File: example/src/admin/admin.ts [hunk 2]
```diff
@@ -1,16 +1,22 @@
 import AdminJSExpress from '@adminjs/express';
-import MongooseAdapter from '@adminjs/mongoose';
-import AdminJS from 'adminjs';
+import AdminJS, { AdminJSOptions } from 'adminjs';
 import { Express } from 'express';
+import path from 'path';
+import { themes } from './themes';
 
-import { createUserResource } from './resources/user/user.resource';
+const PALETE_PAGE = AdminJS.bundle(path.join(__dirname, './components/palete'));
```
Comment (dziraf) [inline example/src/admin/admin.ts:7]: `PALETE` vs `PALETTE`?

File: package.json [hunk 3]
```diff
@@ -46,7 +47,9 @@
   },
   "dependencies": {
     "@adminjs/mongoose": "^2.0.0",
```
Comment (dziraf) [inline package.json:49]: I think it's better to keep example app dependencies in it's separate package.json

File: src/theme-generator.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+import chroma from 'chroma-js';
+import keys from 'lodash/keys';
+import merge from 'lodash/merge';
+
+type PaleteColors = Partial<Record<'primary' | 'grey', string>>;
```
Comment (dziraf) [inline src/theme-generator.ts:7]: `Palete` vs `Palette`?

File: src/themes/dark/dark-theme.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { AdminJSTheme } from 'adminjs';
+import ThemeGenerator from '../../theme-generator';
+
+const DarkThemeOptions: AdminJSTheme = {
```
Comment (dziraf) [inline src/themes/dark/dark-theme.ts:4]: Is it possible to override i. e. logo via theme? Some logos look good on white backgrounds but won't look good on dark background so we'd need a way to specify separate logos for each theme.

File: example/src/admin/admin.ts [hunk 6]
```diff
@@ -1,16 +1,22 @@
...
 
-const setupAdmin = async (app: Express): Promise<void> => {
-  AdminJS.registerAdapter(MongooseAdapter);
-  const adminJs = new AdminJS({
-    resources: [createUserResource()],
-  });
+export const adminConfig: AdminJSOptions = {
+  brandings: themes.map(theme => ({ theme })),
```
Comment (dziraf) [inline example/src/admin/admin.ts:10]: Isn't it `availableBrandings`? https://github.com/SoftwareBrothers/adminjs/pull/1025/files#diff-dfa882cb9ff0eb0aaf1f860e243a9cf99c29bd58925c7d451ebb2a2f78c98566R125

File: example/src/admin/admin.ts (hunk 2 above)
Comment (ariansobczak-rst) [reply example/src/admin/admin.ts]: Renamed

File: src/theme-generator.ts (hunk 4 above)
Comment (ariansobczak-rst) [reply src/theme-generator.ts]: Renamed

File: example/src/admin/admin.ts (hunk 6 above)
Comment (ariansobczak-rst) [reply example/src/admin/admin.ts]: Good catch! True should be `availableBrandings`

**Review by dziraf — APPROVED** [review]

File: src/themes/dark/dark-theme.ts (hunk 5 above)
Comment (ariansobczak-rst) [reply src/themes/dark/dark-theme.ts:4]: I wonder to keep logo responsible for branding object and configure brandings like 
```ts
  availableBrandings: [
    {
      theme: LightTheme,
      logo: '/logo.svg',
    },
    {
      theme: DarkTheme,
      logo: '/logo_white.svg',
    },
  ],
```
and the main branding like e.g.
```ts
  branding: {
    theme: DarkTheme,
    logo: '/logo_white.svg',
  },
```

Comment (ariansobczak-rst) [reply src/themes/dark/dark-theme.ts:4]: I added white logo to example

**Pull request merged**